pkg/maps/lbmap/source_range.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lbmap
+
+import (
+	"fmt"
+	"net"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/cidr"
+	"github.com/cilium/cilium/pkg/types"
+)
+
+const (
+	SourceRange4MapName = "cilium_lb4_source_range"
+	SourceRange6MapName = "cilium_lb6_source_range"
+	lpmPrefixLen4       = 16 + 16 // sizeof(SourceRangeKey4.RevNATID)+sizeof(SourceRangeKey4.Pad)
+	lpmPrefixLen6       = 16 + 16 // sizeof(SourceRangeKey6.RevNATID)+sizeof(SourceRangeKey6.Pad)
```

Comment:
Why not creating 2 types?
```golang
type RevNATID  uint16
type Pad uint16

const(
	lpmPrefixLen4       = unsafe.SizeOf(RevNATID)+unsafe.SizeOf(Pad)
	lpmPrefixLen6       = unsafe.SizeOf(RevNATID)+unsafe.SizeOf(Pad)
)
```

Reply:
Yeah, I thought about it, but then we would need to use those types everywhere in `pkg/lbmap`.

Reply:
Not really no.
```patch
diff --git a/pkg/maps/lbmap/source_range.go b/pkg/maps/lbmap/source_range.go
index cb31b3b38..e9360b729 100644
--- a/pkg/maps/lbmap/source_range.go
+++ b/pkg/maps/lbmap/source_range.go
@@ -28,8 +28,16 @@ import (
 const (
        SourceRange4MapName = "cilium_lb4_source_range"
        SourceRange6MapName = "cilium_lb6_source_range"
-       lpmPrefixLen4       = 16 + 16 // sizeof(SourceRangeKey4.RevNATID)+sizeof(SourceRangeKey4.Pad)
-       lpmPrefixLen6       = 16 + 16 // sizeof(SourceRangeKey6.RevNATID)+sizeof(SourceRangeKey6.Pad)
+)
+
+type RevNATID = uint16
+type Pad = uint16
+
+const (
+       __revNATID    = RevNATID(0)
+       __pad         = Pad(0)
+       lpmPrefixLen4 = uint32(unsafe.Sizeof(__revNATID) + unsafe.Sizeof(__pad))
+       lpmPrefixLen6 = uint32(unsafe.Sizeof(__revNATID) + unsafe.Sizeof(__pad))
 )
 
 type SourceRangeKey interface {
@@ -41,8 +49,8 @@ type SourceRangeKey interface {
 // +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
 type SourceRangeKey4 struct {
        PrefixLen uint32     `align:"lpm_key"`
-       RevNATID  uint16     `align:"rev_nat_id"`
-       Pad       uint16     `align:"pad"`
+       RevNATID  RevNATID   `align:"rev_nat_id"`
+       Pad       Pad        `align:"pad"`
        Address   types.IPv4 `align:"addr"`
 }
 
@@ -61,11 +69,12 @@ func (k *SourceRangeKey4) GetCIDR() *cidr.CIDR {
                cidr cidr.CIDR
                ip   types.IPv4
        )
-       cidr.Mask = net.CIDRMask(int(k.PrefixLen)-lpmPrefixLen4, 32)
+       cidr.Mask = net.CIDRMask(int(k.PrefixLen-lpmPrefixLen4), 32)
        k.Address.DeepCopyInto(&ip)
        cidr.IP = ip.IP()
        return &cidr
 }
+
 func (k *SourceRangeKey4) GetRevNATID() uint16 {
        return k.RevNATID
 }
@@ -94,7 +103,7 @@ func (k *SourceRangeKey6) GetCIDR() *cidr.CIDR {
                cidr cidr.CIDR
                ip   types.IPv6
        )
-       cidr.Mask = net.CIDRMask(int(k.PrefixLen)-lpmPrefixLen6, 128)
+       cidr.Mask = net.CIDRMask(int(k.PrefixLen-lpmPrefixLen6), 128)
        k.Address.DeepCopyInto(&ip)
        cidr.IP = ip.IP()
        return &cidr
```

Reply:
I don't want to introduce new types in the package which will be used partially. I have in my TODO to introduce lbmap types, and replace all `u{8,16,32}` with them which would address your suggestion.